runtime/gc_include/ObjectAccessBarrierAPI.hpp

```diff
@@ -395,8 +388,11 @@ class MM_ObjectAccessBarrierAPI
 			}
 		} else {
 			UDATA offset = 0;
-			bool isValueType = J9_IS_J9CLASS_VALUETYPE(objectClass);
-			UDATA limit = mixedObjectGetDataSize(objectClass, isValueType);
+			UDATA limit = mixedObjectGetDataSize(objectClass);
+			
+			if (J9_IS_J9CLASS_VALUETYPE(objectClass)) {
+				offset += objectClass->backfillOffset;
```

Comment:
`backfillOffset` is the start of the first field for value types?

Reply:
if so can we just make it `offset = objectClass->backfillOffset` as I think that is clearer.  Anyone disagree?

Reply:
I don't think two assignments is clearer (though clearly more efficient), unless there's clearly in an if/else or ternary path.

Reply:
My concern is that if someone changes the default value of the `offset` variable we may do the incorrect thing by accident.  Another suggestion then.  Since offset is always 0 for non value types and backfillOffset for value types maybe should just add backfillOffset to [src,dest]Offset when it is a value type?

Reply:
Assigning would have the opposite issue (if offset were meaningful before the VT test, an assignment would lose any previous computation).

As the VT test is only doing this addition, would a ternary assignment be better for you?

Reply:
Hmmm I don't really like the ternary operations lol.  We discussed it and there is technically nothing wrong with the current code so I say just leave it.